MTFO/Custom/Scans.cs

```diff
@@ -9,32 +10,42 @@ public class ScanHolder
         public List<CustomClusterScan> Clusters { get; set; }
     }
 
+    public enum RevealDistanceMode
+    {
+        Distance,
+        Time
+    }
+
     public struct CustomBioScan
     {
         public uint BaseScan { get; set; }
         public uint PersistentID { get; set; }
-        public int PlayerRequirement { get; set; }
+        public PlayerRequirement PlayerRequirement { get; set; }
         public float ScanRadius { get; set; }
         public float[] PlayersInScanMulti { get; set; }
         public float ReduceSpeed { get; set; }
         public bool ReduceWhenNoPlayer { get; set; }
+        public float RevealTime { get; set; }
+        [JsonIgnore]
+        internal float SplineRevealSpeed => 1f / this.RevealTime;
+        public RevealDistanceMode RevealTimeDistanceMode { get; set; }
         public BioScanGx BioScanGraphics { get; set; }
 
 
         public struct BioScanGx
         {
             public string ScanText { get; set; }
             public float Radius { get; set; }
-            public BioScanColorByMode[] colorModeColor { get; set; }
+            public BioScanColorByMode[] ColorModeColor { get; set; }
```

Comment:
Wouldn't this break current configuration?

Reply:
Case-Insensitive deserialization